sql/core/src/main/scala/org/apache/spark/sql/scripting/SqlScriptingExecutionNode.scala

```diff
@@ -102,10 +122,11 @@ class SingleStatementExec(
 abstract class CompoundNestedStatementIteratorExec(collection: Seq[CompoundStatementExec])
   extends NonLeafStatementExec {
 
-  private var localIterator = collection.iterator
-  private var curr = if (localIterator.hasNext) Some(localIterator.next()) else None
+  protected var localIterator: Iterator[CompoundStatementExec] = collection.iterator
+  protected var curr: Option[CompoundStatementExec] =
+    if (localIterator.hasNext) Some(localIterator.next()) else None
 
-  private lazy val treeIterator: Iterator[CompoundStatementExec] =
+  protected lazy val treeIterator: Iterator[CompoundStatementExec] =
```

Comment:
let's remove `treeIterator` implementation from `CompoundNestedStatementIteratorExec` for now, and let's then figure out offline if we need `CompoundNestedStatementIteratorExec` at all in the future